internal/encoding/parser.go

```diff
@@ -0,0 +1,90 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/encoding/graphite"
+)
+
+type Parser struct {
+	graphiteParser *graphite.Parser
+}
+
+func NewParser(parser *graphite.Parser) *Parser {
+	return &Parser{graphiteParser: parser}
+}
+
+func (p *Parser) Parse(dataFormat string, out []byte, acc telegraf.Accumulator) error {
```

Comment:
`Parse` functions should _not_ be specific to any plugin, just make two functions here with signatures:

```
func (p *Parser) Parse(dataFormat string, buf []byte) ([]telegraf.Metric, error)
func (p *Parser) ParseLine(dataFormat, line string) (telegraf.Metric, error)
```